server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -275,6 +277,17 @@ class LivyServer extends Logging {
       case _ => false
     }
     require(supportedVersion, s"Unsupported Spark version $version.")
+
+    // Spark >= 1.6 require for Livy yarn-master mode
+    if (sparkMaster == "yarn-master") {
+      val supportedVersion = version match {
+        case versionPattern(major, minor) =>
+          val v = (major.toInt, minor.toInt)
+          v >= minYarnMasterVersion
+        case _ => false
+      }
+      require(supportedVersion, s"YARN master mode unsupported in Spark version $version.")
```

Comment:
I think instead of this it would be better to fall back to using `SparkProcApp` instead of `SparkYarnApp` if version is older than 1.6. But maybe there's something I'm missing that @tc0312 can comment on.